src/hotspot/share/opto/superword.cpp

```diff
@@ -519,8 +462,11 @@ void SuperWord::mark_reductions() {
 //    extraction of scalar values from vectors.
 //
 bool SuperWord::SLP_extract() {
-  CountedLoopNode* cl = lpt()->_head->as_CountedLoop();
-  assert(cl->is_main_loop(), "SLP should only work on main loops");
+  assert(cl()->is_main_loop(), "SLP should only work on main loops");
+
+  if (SuperWordReductions) {
+    mark_reductions();
+  }
```

Comment:
Note: it would be really nice to move reduction marking to all other analysis, like bb construction and velt computation, etc.